File: 프로그래머스/lv3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    static int answer;
    static boolean[] visited;
    public int solution(int n, int[][] computers) {
        answer = 0;
        visited = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (visited[i]) continue;
            bfs(i, computers);
            answer++;
        }
        return answer;
    }
    
    public static void bfs(int st, int[][] computers) {
        Queue<Integer> queue = new ArrayDeque<>();
        queue.add(st);
        visited[st] = true;
        while (!queue.isEmpty()) {
            int curr = queue.poll();
            for (int i = 0; i < computers[curr].length; i++) {
                if (computers[curr][i] == 1 && visited[i] == false) {
                    visited[i] = true;
                    queue.add(i);
                }
            }
        }
    }
}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    
    let a = Array(n).fill(0);
    
    
    const dfs = (index, visited) => {
        if (visited[index] != 0) return; //방문했으면 방문 X
        visited[index] = 1;
        computers[index].forEach((e, i) => {
            if (e == 1) {
                dfs(i, visited);
            }
        });
    }
    
    let count = 0;
    for (let i = 0; i < n; i++) {
        if (a[i] == 0) {
            dfs(i, a)
            count++;
        }
    }
    // console.log(count);
    var answer = count;
    return answer;
}